### PR TITLE
Closes #8402 - Dismiss Select Login Prompt on Refresh

### LINKED ISSUE #8402: Select Login Prompt Not Dismissed on Refresh
Reported https://github.com/mozilla-mobile/fenix/issues/14585

### PR DESCRIPTION
**Pull request by ekager**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt [hunk 1]
```diff
@@ -243,6 +243,10 @@ class PromptFeature private constructor(
                             onPromptRequested(state)
                         } else if (!it.loading) {
                             promptAbuserDetector.resetJSAlertAbuseState()
+                        } else if (it.loading) {
+                            if (activePromptRequest is SelectLoginPrompt) {
+                                loginPicker?.dismissCurrentLoginSelect(activePromptRequest as SelectLoginPrompt)
+                            }
```
Comment (Amejia481) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt:249]: Could we add a test for this use case ?

File: components/concept/engine/src/test/java/mozilla/components/concept/engine/prompt/PromptRequestTest.kt [hunk 2]
```diff
@@ -161,17 +161,19 @@ class PromptRequestTest {
         confirmRequest.onConfirmNegativeButton(true)
         confirmRequest.onConfirmNeutralButton(true)
 
-        val onConfirmLogin: () -> Unit = {}
-        val onDismissLogin: (Login) -> Unit = {}
+        val onLoginDismiss: () -> Unit = {}
```
Comment (ekager) [inline components/concept/engine/src/test/java/mozilla/components/concept/engine/prompt/PromptRequestTest.kt:164]: I just noticed these were named incorrectly in another test so fixed it

**Comment by Amejia481** [comment]
The failure on `build-feature-resentlyclosed` seems unrelated I filed https://github.com/mozilla-mobile/android-components/issues/8410 for it.

**Review by Amejia481 — APPROVED** [review]
🚢

**Comment by ekager** [comment]
bors r=Amejia481

**Pull request merged**